libkbfs/folder_branch_ops.go

```diff
@@ -5447,31 +5448,54 @@ func (fbo *folderBranchOps) backgroundFlusher(betweenFlushes time.Duration) {
 			// so don't bother waiting for a signal, just get right to
 			// the main attraction.
 			doSelect = false
+		} else if fbo.getCachedDirOpsCount(lState) >=
+			fbo.config.BGFlushDirOpBatchSize() {
+			doSelect = false
 		}
 
 		if doSelect {
 			// Wait until we really have a write waiting.
 			forced := false
 			select {
 			case <-fbo.syncNeededChan:
+				if fbo.getCachedDirOpsCount(lState) >=
+					fbo.config.BGFlushDirOpBatchSize() {
+					forced = true
+				}
 			case <-fbo.forceSyncChan:
 				forced = true
 			case <-fbo.shutdownChan:
 				return
 			}
 
 			if !forced {
-				select {
-				case <-ticker.C:
-				case <-fbo.forceSyncChan:
-				case <-fbo.shutdownChan:
-					return
+				timer := time.NewTimer(fbo.config.BGFlushPeriod())
```

Comment:
`time.Timer` discards unreceived ticks. What do you think about keeping a `timer` outside the out-most for loop so that we don't have to keep constructing new ones?